### PR TITLE
Add `spice refresh` command

### PR DESCRIPTION
**Pull request by sgrebnov**
Implements: https://github.com/spiceai/spiceai/issues/1095

Add `spice refresh dataset-name` command support:

Examples: 

1. Dataset exists
```bash
⚡ spice refresh taxi_trips                     
Refreshing dataset taxi_trips ...
Dataset refresh triggered for taxi_trips.
```

1. Dataset does NOT exist
```bash
⚡ spice refresh ny_taxi_trips 
Refreshing dataset ny_taxi_trips ...
Dataset ny_taxi_trips not found
```

1. Runtime is unavailable
```bash
⚡ spice refresh taxi_trips    
Refreshing dataset taxi_trips ...
The Spice runtime is unavailable at http://127.0.0.1:3000. Is it running?
```

### CODE REVIEW

**Review by lukekim — CHANGES_REQUESTED** [review]
Please check off the items in https://github.com/spiceai/spiceai/issues/1095 before starting implementation, including PM/UX sign off by Aurash.

**Comment by sgrebnov** [comment]
> Please check off the items in #1095 before starting implementation, including PM/UX sign off by Aurash.

Thank you - updated so it is `spice refresh` instead of `spice dataset refresh`

File: bin/spice/cmd/refresh.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	Args:  cobra.MinimumNArgs(1),
+	Example: `
+spice refresh taxi_trips
+
+# See more at: https://docs.spiceai.org/
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		dataset := args[0]
+
+		cmd.Printf("Initiating refresh for dataset %s ...\n", dataset)
```
Comment (digadeesh) [inline bin/spice/cmd/refresh.go]: Lets just have it say "Refreshing dataset <dataset>"

File: bin/spice/pkg/api/util.go [hunk 2]
```diff
@@ -26,44 +26,62 @@ import (
...
-			return rtcontext.RuntimeUnavailableError()
-		}
-		return fmt.Errorf("Error fetching %s: %w", url, err)
+		return nil, err
 	}
-	defer resp.Body.Close()
+	return result, nil
+}
+
+func DoRuntimePostRequest[T interface{}](rtcontext *context.RuntimeContext, path string) (T, error) {
```
Comment (digadeesh) [inline bin/spice/pkg/api/util.go]: is the naming a convention you're following?  Previously it was GetData ->. retrieving data from the APIs.  You could just modify that use the appropriate http verb.

File: bin/spice/cmd/refresh.go (hunk 1 above)
Comment (sgrebnov) [reply bin/spice/cmd/refresh.go]: Updated - thank you

File: bin/spice/pkg/api/util.go (hunk 2 above)
Comment (sgrebnov) [reply bin/spice/pkg/api/util.go]: Changed to `api.PostRuntime` - thank you

**Review by digadeesh — APPROVED** [review]

**Review by lukekim — APPROVED** [review]

**Pull request merged**